File: JebraAzureFunctions/JebraAzureFunctions/Models/InstructorModel.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace JebraAzureFunctions.Models
{
    class InstructorModel : Attribute
    {
        [JsonProperty]
        public int id { get; set; }
        [JsonProperty]
        public string fname { get; set; }
        [JsonProperty]
        public string lname { get; set; }
        [JsonProperty]
        public string username { get; set; }
        [JsonProperty]
        public string pass { get; set; }
        [JsonProperty]
        public string email { get; set; }
    }
}

File: client-app/src/models/InstructorModel.ts
// Interface for Instructor JSON data returned from backend
export default interface InstructorModel {
    id: number,
    fname: string,
    lname: string,
    username: string,
    pass: string,
    email: string
}

// Type guard for validating that data returned from the backend contains the expected fields
export function isInstructorModel(data: any): data is InstructorModel {
    if (typeof data !== "object" || Array.isArray(data) || data === null) {
        return false;
    }

    const instructorData = data as InstructorModel;
    return typeof instructorData.id === "number"
        && typeof instructorData.fname === "string"
        && typeof instructorData.lname === "string"
        && typeof instructorData.username === "string"
        && typeof instructorData.pass === "string"
        && typeof instructorData.email === "string";
}